servers/rendering/renderer_rd/storage_rd/texture_storage.cpp

```diff
@@ -2458,6 +2458,27 @@ void TextureStorage::decal_set_texture(RID p_decal, RS::DecalTexture p_type, RID
 	decal->textures[p_type] = p_texture;
 
 	if (decal->textures[p_type].is_valid()) {
+#ifdef TOOLS_ENABLED
+		Texture *tex = get_texture(p_texture);
+		if (tex->detect_3d_callback) {
+			tex->detect_3d_callback(tex->detect_3d_callback_ud);
+		}
+		if (tex->detect_normal_callback && p_type == RS::DecalTexture::DECAL_TEXTURE_NORMAL) {
+			tex->detect_normal_callback(tex->detect_normal_callback_ud);
+		}
+
+		Texture *normal_tex = nullptr;
+		if (decal->textures[RS::DecalTexture::DECAL_TEXTURE_NORMAL].is_valid()) {
+			normal_tex = get_texture(decal->textures[RS::DecalTexture::DECAL_TEXTURE_NORMAL]);
+		}
+		Texture *orm_tex = nullptr;
+		if (decal->textures[RS::DecalTexture::DECAL_TEXTURE_ORM].is_valid()) {
+			orm_tex = get_texture(decal->textures[RS::DecalTexture::DECAL_TEXTURE_ORM]);
+		}
+		if (orm_tex && orm_tex->detect_roughness_callback && normal_tex && !normal_tex->path.is_empty()) {
+			orm_tex->detect_roughness_callback(orm_tex->detect_roughness_callback_ud, normal_tex->path, RS::TEXTURE_DETECT_ROUGHNESS_G);
```

Comment:
Is it correct to assume here that the roughness channel is always green? It's always an ORM texture.

Reply:
It should be fine here, yes.